lib/Sema/CSBindings.cpp

```diff
@@ -204,14 +204,28 @@ ConstraintSystem::getPotentialBindings(TypeVariableType *typeVar) {
 
     case ConstraintKind::BridgingConversion:
     case ConstraintKind::CheckedCast:
-    case ConstraintKind::DynamicTypeOf:
     case ConstraintKind::EscapableFunctionOf:
     case ConstraintKind::OpenedExistentialOf:
     case ConstraintKind::KeyPath:
     case ConstraintKind::KeyPathApplication:
       // Constraints from which we can't do anything.
       continue;
 
+    case ConstraintKind::DynamicTypeOf: {
+      // If given type variable is the left-hand side of
+      // the dynamic type constraint, let's not attempt
+      // to bind it directly since right-hand side is
+      // supposed to do just that. It's also not always safe
+      // to do since it might become bound to something with
+      // is not [existential] metatype but still be convertible
```

Comment:
"that is not an existential metatype but is still convertible to one" maybe?

However that's still slightly wrong, we're not converting Any to a metatype, we're converting a metatype to Any.

And Any is the *only* existential that has metatypes as subtypes of itself, AFAIK.